### PR TITLE
build(deps): bump react-native from 0.73.10 to 0.75.4

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Bumps `react-native` version from `0.73.10` to `0.75.4`.

#### Removed Patches:

- [x] react-native-mapbox-gl/maps 
- [x] react-native 
- [x] deprecated-react-native-prop-types dep (included in the react native patch)

#### Bumped Dependencies:

- `@gorhom/bottom-sheet`
- `@react-native-async-storage/async-storage`
- `"@react-native-mapbox-gl/maps": "8.5.0"` removed in favor of `"@rnmapbox/maps": "10.1.31",`
- `@shopify/flash-list` 
- `@stripe/stripe-react-native` 
- `moti`
- `react-native-reanimated`

#### Removed Dependencies:

Flipper related stuff and react devtools (the react dev tools are still acessible with `npx react-devtools`

- `"deprecated-react-native-prop-types": "4.1.0",`
- `"react-devtools-core"`
- `"rn-flipper-async-storage-advanced": "1.0.5",`
- `"redux-flipper": "2.0.2",`
- `"react-native-flipper": "0.265.0",`
- `"react-native-flipper-performance-plugin": "0.4.0",`
- `"react-native-flipper-relay-devtools": "1.0.3",`

> [!IMPORTANT]  
> Flipper is decoupled from react native - we will run a survey to see if we need to bring it back or not

#### 

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- bump react-native version

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/utils/useFreshInstallTracking.ts [hunk 1]
```diff
@@ -6,9 +6,12 @@ import { postEventToProviders } from "./track/providers"
 export const useFreshInstallTracking = () => {
   useEffect(() => {
     const launchCount = getCurrentEmissionState().launchCount
+    console.warn("[debug]", { launchCount })
```
Comment (gkartalis) [inline src/app/utils/useFreshInstallTracking.ts:9]: these are meant for testing - will remove

File: src/app/NativeModules/LegacyNativeModules.tsx [hunk 2]
```diff
@@ -127,11 +128,10 @@ const LegacyNativeModulesAndroid = {
   },
 
   ARNotificationsManager: {
-    // nativeState: null as any,
     postNotificationName: noop("postNotificationName"),
     didFinishBootstrapping: () => null,
     reactStateUpdated: () => null,
-    getConstants: () => ({}),
+    getConstants: () => null as any,
```
Comment (gkartalis) [inline src/app/NativeModules/LegacyNativeModules.tsx:134]: I went with this as it was in line 130 to avoid solving extra type issues but it is not ideal

**Comment by gkartalis** [comment]
ios-8.56.0-2024.11.07.09

android-8.56.0-1674645923

File: package.json [hunk 3]
```diff
@@ -104,40 +104,38 @@
   ],
   "dependenciesComments": {
     "react-native": "Try running `npx @rnx-kit/align-deps --requirements react-native@0.66.5` when upgrading to a version, here 0.66.5",
-    "deprecated-react-native-prop-types": "Needed for patching react-native due to rn removing the proptypes from their repo",
```
Comment (gkartalis) [inline package.json:107]: 👋

File: HACKS.md [hunk 4]
```diff
@@ -38,38 +38,6 @@ When we upgrade to a version of `@segment/analytics-react-native` that includes
...
-## deprecated-react-native-prop-types dependency
-
-#### When can we remove this:
-
-When we upgrade our deps to a version of react-native that includes removal of deprecated PropTypes.
-
-#### Explanation/Context:
-
-When updating to rn-0.69.10 we had to patch this due to deprecation of PropTypes. For this reason we also installed `deprecated-react-native-prop-types` to avoid errors and we patched the `react-native` package to use the deprecated PropTypes coming from the `deprecated-react-native-prop-types` package.
-
```
Comment (brainbicycle) [inline HACKS.md:72]: 🎉

File: ios/Artsy/Emission/ViewControllers/ARMapContainerViewController.m [hunk 5]
```diff
@@ -265,16 +263,23 @@ - (void)showCityPicker
...
-{
-    RCTMGLMapView *mapView = FindMapView(self.mapVC.view);
-
-    if (!self.attributionViewsConstraintsAdded) {
-        [mapView.attributionButton alignBottomEdgeWithView:self.mapVC.view predicate:@"-50"];
-        [mapView.logoView alignBottomEdgeWithView:self.mapVC.view predicate:@"-50"];
-        self.attributionViewsConstraintsAdded = YES;
-    }
-}
+// TODO: display attribution on rn side
```
Comment (brainbicycle) [inline ios/Artsy/Emission/ViewControllers/ARMapContainerViewController.m:266]: oh we may have forgot about this one 👁️

File: ios/Podfile [hunk 6]
```diff
@@ -19,20 +19,9 @@ inhibit_all_warnings! # ignore all warnings from all pods
...
-# If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
-# because `react-native-flipper` depends on (FlipperKit,...) that will be excluded
-#
-# To fix this you can also exclude `react-native-flipper` using a `react-native.config.js`
-# ```js
-# module.exports = {
-#   dependencies: {
-#     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
-# ```
-flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled
```
Comment (brainbicycle) [inline ios/Podfile:35]: 🐬 🙅

File: src/app/AppRegistry.tsx [hunk 7]
```diff
@@ -708,9 +721,13 @@ export const modules = defineModules({
 })
 
 // Register react modules with the app registry
+registerCity()
 for (const moduleName of Object.keys(modules)) {
   const descriptor = modules[moduleName as AppModule]
   if (Platform.OS === "ios") {
+    if (moduleName === "City") {
+      console.warn("Registering City moudle for iOS")
+    }
```
Comment (brainbicycle) [inline src/app/AppRegistry.tsx:730]:
```suggestion
```

File: src/app/AppRegistry.tsx [hunk 8]
```diff
@@ -708,9 +721,13 @@ export const modules = defineModules({
 })
 
 // Register react modules with the app registry
+registerCity()
```
Comment (brainbicycle) [inline src/app/AppRegistry.tsx:724]: TODO: revisit this, is it necessary?

File: src/app/Components/Containers/Inbox.tsx [hunk 9]
```diff
@@ -49,6 +49,7 @@ export class Inbox extends React.Component<Props, State> {
   flatListHeight = 0
 
   componentDidMount() {
+    // @ts-ignore
```
Comment (brainbicycle) [inline src/app/Components/Containers/Inbox.tsx:52]: not sure if worth a timebox to fix types

File: src/app/NativeModules/CityGuideView.tsx [hunk 10]
```diff
@@ -1,9 +1,9 @@
 import { requireNativeComponent } from "react-native"
 
-const ARTCityGuideView = requireNativeComponent("ARTCityGuideView")
+const ARTMap = requireNativeComponent("ARTCityGuideView")
```
Comment (brainbicycle) [inline src/app/NativeModules/CityGuideView.tsx:3]: think I renamed this when we were trying to debug what was wrong, will change back

File: src/app/Scenes/Map/Components/PinsShapeLayer.tsx [hunk 11]
```diff
@@ -11,7 +11,7 @@ import { Animated, StyleProp } from "react-native"
 
 interface Props {
   featureCollections: { [key in BucketKey]: FilterData } | {}
-  onPress?: ShapeSourceProps["onPress"]
+  onPress?: any["onPress"]
```
Comment (brainbicycle) [inline src/app/Scenes/Map/Components/PinsShapeLayer.tsx:14]: possibly worth revisit

File: src/app/Scenes/Map/GlobalMap.tsx [hunk 12]
```diff
@@ -604,7 +606,7 @@ export class GlobalMap extends React.Component<Props, State> {
                       <PinsShapeLayer
                         filterID={cityTabs[this.state.activeIndex].id}
                         featureCollections={this.state.featureCollections}
-                        onPress={(e) => this.handleFeaturePress(e)}
+                        onPress={(e: any) => this.handleFeaturePress(e)}
```
Comment (brainbicycle) [inline src/app/Scenes/Map/GlobalMap.tsx:609]: revisit type

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 70330d4777df37d06a06062789d34a6d48745e2a
</p>

File: src/app/AppRegistry.tsx [hunk 13]
```diff
@@ -286,6 +292,13 @@ function register(
   AppRegistry.registerComponent(screenName, () => WrappedComponent)
 }
 
+// function registerCity() {
+//   console.warn("Registering City")
+//   AppRegistry.registerComponent("City", () => CityView)
+//   AppRegistry.registerComponent("Map", () => MapContainer)
+//   AppRegistry.registerComponent("CityPicker", () => CityPicker)
+// }
+
```
Comment (brainbicycle) [inline src/app/AppRegistry.tsx:301]:
```suggestion
```

File: src/app/AppRegistry.tsx [hunk 14]
```diff
@@ -275,6 +275,12 @@ class PageWrapper extends React.Component<PageWrapperProps> {
   }
 }
 
+/**
+ * eigen brigde
+ * (RCTBridge *) 0x0000600001cd17a0
+ * (RCTBridge *) 0x0000600001ccc870
+ */
+
```
Comment (brainbicycle) [inline src/app/AppRegistry.tsx:283]:
```suggestion
```

**Comment by MounirDhahri** [comment]
Impressive effort 👏👏👏

**Pull request merged**